platform/security/idp/security-idp-plugin/security-idp-plugin-audiencerestriction/src/test/groovy/org/codice/ddf/security/idp/plugin/audiencerestriction/AudienceRestrictionPresignPluginSpec.groovy

```diff
@@ -0,0 +1,131 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.security.idp.plugin.audiencerestriction
+
+import ddf.security.samlp.SamlProtocol
+import org.apache.wss4j.common.saml.OpenSAMLUtil
+import org.opensaml.saml.saml2.core.*
+import spock.lang.Specification
+
+class AudienceRestrictionPresignPluginSpec extends Specification {
+    def ISSUER_ID = "https://localhost:8993/services/idp/login"
+
+    // general mocks
+    AudienceRestrictionPresignPlugin plugin = new AudienceRestrictionPresignPlugin()
+    List<String> spMetadata = new ArrayList<>()
+    Set<SamlProtocol.Binding> bindings = new HashSet<>()
+    AuthnRequest authNReq = Mock(AuthnRequest)
+    Issuer issuer = Mock(Issuer)
+
+    // response mocks
+    Response response = Mock(Response)
+    List<Assertion> assertions = new ArrayList<>()
```

Comment:
✏️ Same: non-mock supporting objects would probably be clearer moved to a separate section from the mock generations.

Reply:
I would like to keep it this way; these are in hierarchical order. The top most one requires the second top most one and so on. The only exception is near the bottom where the numbers represent siblings.

Reply:
Understood. Then I suggest at least changing your in-line comments to reflect that this is a heterogenous set of mocks and concrete objects.